timeutil/schedule.go

```diff
@@ -82,40 +108,285 @@ func ParseTime(s string) (t TimeOfDay, err error) {
 	return t, nil
 }
 
-// Schedule defines a start and end time in which events should run.
-type Schedule struct {
+// Weekday represents a weekday such as Monday, Tuesday, with optional
+// week-in-the-month position, eg. the first Monday of the month
+type Weekday struct {
+	// Day stands for a weekday - "mon", "tue", ..
+	Day string
+	// Week the day appears in given month, 5 means the last week
+	Pos uint
+}
+
+// IsZero indicates if Weekday is in the default state
+func (w Weekday) IsZero() bool {
+	return w == Weekday{}
+}
+
+func (w Weekday) String() string {
+	if w.Pos == 0 {
+		return w.Day
+	}
+	return w.Day + strconv.FormatUint(uint64(w.Pos), 10)
```

Comment:
(nitpick) you could use `strconv.Itoa(int(w.Pos))` instead of FormatUint() which is slightly shorter.